sunpy/map/sources/yohkoh.py

```diff
@@ -57,23 +56,22 @@ def detector(self):
     @property
     def dsun(self):
         """
-        For Yohkoh Maps, DSUN_OBS is not always defined. In this case the
-        SOLAR_R keyword is used to calculate dsun.
-        """
+        For Yohkoh Maps, DSUN_OBS is not always defined.
 
+        In this case the SOLAR_R keyword is used to calculate dsun.
+        """
         # 2012/12/19 - the SXT headers do not have a value of the distance from
-        # the spacecraft to the center of the Sun.  The FITS keyword 'DSUN_OBS'
-        # appears to refer to the observed diameter of the Sun.  Until such
+        # the spacecraft to the center of the Sun. The FITS keyword 'DSUN_OBS'
+        # appears to refer to the observed diameter of the Sun. Until such
         # time as that is calculated and properly included in the file, we will
         # use simple trigonometry to calculate the distance of the center of
-        # the Sun from the spacecraft.  Note that the small angle approximation
+        # the Sun from the spacecraft. Note that the small angle approximation
         # is used, and the solar radius stored in SXT FITS files is in arcseconds.
-
         if 'solar_r' in self.meta:
             dsun = constants.radius / (np.deg2rad(self.meta['solar_r'] / 3600.0))
         else:
             dsun = constants.au
-        return self.meta.get('dsun_obs', dsun)
+        return dsun
```

Comment:
To align with the comment, I decided to calculate the value and return that instead.

Reply:
Well...  The test header has a usable (and presumably) correct value for `DSUN_OBS`...

Reply:
I think we should get rid of this overridden `dsun` property entirely.  As currently implemented, `.dsun` is not used to construct the observer coordinate, so the change of inconsistency is high.  If `DSUN_OBS` is missing, we need `.dsun` to give the same answer as `.observer_coordinate.radius`.  If `DSUN_OBS` is present but wildly wrong, that error needs to be addressed during the creation of the observer coordinate, not in this property.

Reply:
This is all gone now and using your suggested fixes.